TCF update: deprecate global scope, OOB (#301)
InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework: TCFv2/IAB Tech Lab - Consent string and vendor list formats v2.md

A TC String that contains positive consent signals must not be created before clear affirmative action is taken by a user that unambiguously signifies that user’s consent. However, a TC String may be created with only legitimate interest establishment signals providing that legitimate interest transparency has been established in accordance with the [Policies](https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/).

### What is the scope for a TC String?

> CMPs must be set up to operate in a **service-specific**. A  TC String in this context is only used by the site(s) or app(s) on which it is running. One is created for every user on a given site/app or group of sites/apps. They may contain [Publisher restrictions](#what-are-publisher-restrictions), a _**[Publisher TC](#publisher-purposes-transparency-and-consent)**_ segment and an _**[AllowedVendors](#allowed-vendors-oob)**_ segment.

Reviewer: remove `be set up to`, and add the word "scope" after `**service-specific**`

A TC String that contains positive consent signals must not be created before clear affirmative action is taken by a user that unambiguously signifies that user’s consent. However, a TC String may be created with only legitimate interest establishment signals providing that legitimate interest transparency has been established in accordance with the [Policies](https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/).

### What is the scope for a TC String?

> CMPs must be set up to operate in a **service-specific** or group-specific configuration. A TC String in this context is applicable only on a service or group of services, for example the site(s) or app(s) on which it is running. One is created for every user on a given site/app or group of sites/apps. They may contain _**[Publisher restrictions](#what-are-publisher-restrictions)**_ and a _**[Publisher TC](#publisher-purposes-transparency-and-consent)**_ segment when returned by the CMP API.

Reviewer 2: I think it would be helpful to have a short statement here that acknowledges there once was a global scope, what its function was, and that it was deprecated September 1, 2021.

Reviewer 3: Agree and propose:
What happened to Global scope and Out-of-Band ?

The TCF Policy previously allowed legal bases in the Framework to be established with global scope, which means a legal basis is not only applicable on the service or group of services (service-specific and group-specific scopes) where it is obtained and managed, but all services implementing global scope preferences. In the context of global scope, TC Strings were stored in a 3rd party cookie associated with the “consensu.org” domain that enabled Consent Management Providers (CMPs) to read the “TC strings from their subdomains [cmp-name].mgr.consensu.org across different services. Deprecation of global-scope support was announced on June 22nd 2021. Alongside the deprecation of global scope, support for Out-of-Band (OOB) - which refers to instances where a legal basis has not been established using the TCF in a global-scope context - was also deprecated. Since Sept 1st 2021, TC strings established with global-scope are considered invalid."

Reviewer 2: Added

### What is the scope for a TC String?

> CMPs must be set up to operate in a **service-specific** or group-specific configuration. A TC String in this context is applicable only on a service or group of services, for example the site(s) or app(s) on which it is running. One is created for every user on a given site/app or group of sites/apps. They may contain [Publisher restrictions](#what-are-publisher-restrictions) and a _**[Publisher TC](#publisher-purposes-transparency-and-consent)**_ segment when returned by the CMP API.

Reviewer 4: Formatting suggestion, I think we are using bold/italic for all the cross-references. 

```suggestion
CMPs must be set up to operate in a **service-specific** or group-specific configuration. A TC String in this context is applicable only on a service or group of services, for example the site(s) or app(s) on which it is running. One is created for every user on a given site/app or group of sites/apps. They may contain _**[Publisher restrictions](#what-are-publisher-restrictions)**_ and a _**[Publisher TC](#publisher-purposes-transparency-and-consent)**_ segment when returned by the CMP API.
```

> There are two main contexts in which a TC String can be created:

Reviewer 5: @anderagakura suggest you resolve this lingering reference to "global" such as this one ("two main contexts") and later "service-specific or global," &c.

Author: @dmdabbs new update (https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/pull/301/commits/48a7e378d969fbf903f23b7187bb95e198f983f8)

**Note:** other personal data, like IP addresses or callee cookies, may be passed as part of the request, and the `gdpr` and `gdpr_consent_xxxxx` is used by the callee to determine whether an identifier cookie or other personal data can be set and/or used.

### What if consent is governed differently in a country?

[Policies](https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/) require consent for Purpose 1 to store and/or access information on a device  “where such consent is necessary” leaving the responsibility to publishers and vendors to determine if consent in those jurisdictions is required or not.

>  If a publisher is operating a CMP within a jurisdiction that does not require consent to store and/or access information on a device and, therefore, does not ask for consent on behalf of a vendor for a purpose 1, the CMP will write the corresponding bit in the _**PurposesConsent**_ field to `0`. Even though it is valid within that jurisdiction to not request consent for Purpose 1, a vendor would interpret that `0` as a “no consent” signal and have no way of knowing that consent was not required in the jurisdiction in which the publisher operates.  This lack of transparency would, ultimately, cause losses in ad revenue for that publisher.

Reviewer 2: This change seems unrelated to the Global Scope change, no?

Author: From Ninon "_Yes indeed this is unrelated to the deprecation of global scope & OOB but it's necessary to make the change as the previous version mentioning legitimate interest does not make sense from a legal perspective. The change remains quite minimal and does not change the explanation as to how this field should be populated._"

### How should a Transparency & Consent String be stored?

In version 2 of the TCF Specifications, the storage mechanism used for service-specific and group-specific TC Strings is up to a CMP, including any non-cookie storage mechanism.

### What are the Purposes and Features being supported?

The IAB Europe Transparency & Consent Framework [Policies](https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/) defines Purposes, Special Purposes, Features, Special Features, and Stacks (groupings of Purposes and/or Special Features). You can reference the details of these purposes and features in the document found at the following URL:

[https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/](https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/)

#### Managing conflicting string versions

> Post 30 September 2020, [v1.x strings were considered invalid](https://iabeurope.eu/all-news/the-iab-europe-transparency-consent-framework-tcf-steering-group-votes-to-extend-technical-support-for-tcf-v1-1/). If a CMP encounters a situation where both a v1.x string and a v2.0 string are erroneously present simultaneously, the CMP should remove the v1.x string to ensure that there is only one source of truth for consumers of the string.

Reviewer 2: Is this new language? It isn't related to the Global Scope change so it seems messy to include it in this update if so.

Author: From Ninon "_Also unrelated to the deprecation of global scope & OOB - although the change is minimal, we just removed "before" by "post" and "will be" by "were" as we are now post September 2020._"

Note: TCF version 2.0 introduces [“Publisher Restrictions”](#what-are-publisher-restrictions), which, if exhausted by a publisher, could result in TC strings that are larger than the size limit for cookies. While this possibility is remote, it should be guarded against – a CMP should work with a publisher to help them accomplish their goals. CMPs may need to take this into consideration when deciding on the storage mechanism for those TC Strings.

### TC String Format

There are 3 distincts TC String segments that are joined together on a “dot” character.  They are:

*   The core vendor transparency and consent details

> *   Disclosed vendors

Reviewer 2: This isn't staying in for signaling purposes, right, @asr-enid?

Reviewer 4: Nope - its just kept here in the listing as this generally lists all segments. The explanation on disclosed vendors makes it explicit that its only usable for storage purposes (optional) - if you'd like on could add

```suggestion
*   Disclosed vendors (optional - not included when returned by CMP API)
```
but honestly that makes it more complicated that needed I guess

### How should a Transparency & Consent String be stored?

> In version 1 of the TCF Specifications the consent string was specified to be stored as either a 1st party cookie for service-specific consent or a 3rd party cookie for global consent. In version 2 of the TCF Specifications, the storage mechanism used for service-specific TC Strings is up to a CMP, including any non-cookie storage mechanism. 

Reviewer: Perhaps mention that previous revisions of version 2 also used a 3rd party cookie for global consent.

### TC String Format

There are 3 distincts TC String segments that are joined together on a “dot” character.  They are:

*   The core vendor transparency and consent details

> *   Disclosed vendors for solving ambiguous negative vendor signal and

Reviewer 4: Should we just leave it with Disclosed vendors (and the explanation in the segment description) for now until we figure out the signalling aspect of this? Given there is no expiation on how to use it for disambiguation that might be misleading for now. 

```suggestion
*   Disclosed vendors
```

    </tr>
  </thead>
  <tbody>

>     <tr>

Reviewer: this table, now with one non-header row, seems to serve little purpose now.

Reviewer 2: Agree. It would make sense to pull the information out of tabular form.

[https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/](https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/)

### TC String Format

There are 2 distincts TC String segments that are joined together on a “dot” character.  They are:

*   The core vendor transparency and consent details and
*   Publisher purposes transparency and consent for their own data uses.

The _**[Core String](#the-core-string)**_ is always required and comes first and includes all the details required for communicating basic vendor transparency and consent. A TC String with all four segments is possible in certain conditions.

```
COw4XqLOw4XqLAAAAAENAXCAAAAAAAAAAAAAAAAAAAAA.IFukWSQgAIQwgI0QEByFAAAAeIAACAIgSAAQAIAgEQACEABAAAgAQFAEAIAAAGBAAgAAAAQAIFAAMCQAAgAAQiRAEQAAAAANAAIAAggAIYQFAAARmggBC3ZCYzU2yIA.QFukWSQgAIQwgI0QEByFAAAAeIAACAIgSAAQAIAgEQACEABAAAgAQFAEAIAAAGBAAgAAAAQAIFAAMCQAAgAAQiRAEQAAAAANAAIAAggAIYQFAAARmggBC3ZCYzU2yIA.YAAAAAAAAAAAAAAAAAA
```

> A service-specific TC String must contain a Core TC String and may optionally contain a _**[Publisher TC](#publisher-purposes-transparency-and-consent)**_ segment :

Reviewer: Perhaps remove "service-specific", as there isn't another kind now.

  </tbody>
</table>

#### Disclosed Vendors

> The _**DisclosedVendors**_ is an optional TC String segment that signals which vendors have been disclosed to a given user by a CMP. It may be used by a CMP while [storing](#how-should-a-transparency--consent-string-be-stored) TC Strings to record which vendors have been disclosed to the user, but must not be included in the TC String when returned by the CMP API.

Reviewer 4: I'd suggest the following such that we do not use 'signal' to avoid any sort of potential misinterpretation, until we proceed on the discussion if and when this segment might be returned by the CMP API in the future

```suggestion
The _**DisclosedVendors**_ is an optional TC String segment that records which vendors have been disclosed to a given user by a CMP. It may be used by a CMP while [storing](#how-should-a-transparency--consent-string-be-stored) TC Strings, but must not be included in the TC String when returned by the CMP API.
```
